#! /bin/sh /usr/share/dpatch/dpatch-run
## stable-042-3834_gphoto2-cameralist.dpatch by Tim Retout <tim@retout.co.uk>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from upstream stable branch r3834 to fix a bug in the gphoto2
## DP: bindings. (Closes: #475662)

@DPATCH@
commit 99209c5ec0da41bbf7c14bad574120b8e41c3b73
Author: sdelcroix <sdelcroix@772769d8-d925-0410-89eb-a6ffa0d40526>
Date:   Tue Apr 15 11:02:19 2008 +0000

    2008-04-15  Stephane Delcroix  <sdelcroix@novell.com>
    
    	backporting r3831 to the 042_STABLE branch
    
    
    git-svn-id: http://svn.gnome.org/svn/f-spot/branches/FSPOT_0_4_2_STABLE@3834 772769d8-d925-0410-89eb-a6ffa0d40526

diff --git a/ChangeLog b/ChangeLog
index 393b6d1..3d1a32e 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,9 @@
 2008-04-15  Stephane Delcroix  <sdelcroix@novell.com>
 
+	backporting r3831 to the 042_STABLE branch
+
+2008-04-15  Stephane Delcroix  <sdelcroix@novell.com>
+
 	backporting r3832 to the 042_STABLE branch
 
 2008-03-25  Stephane Delcroix  <sdelcroix@novell.com>
diff --git a/libgphoto2-sharp/CameraList.cs b/libgphoto2-sharp/CameraList.cs
index dfc5dbb..24e2a6c 100644
--- a/libgphoto2-sharp/CameraList.cs
+++ b/libgphoto2-sharp/CameraList.cs
@@ -69,27 +69,27 @@ namespace LibGPhoto2
 		}
 		
 		[DllImport ("libgphoto2.so")]
-		internal static extern ErrorCode gp_list_get_name (HandleRef list, int index, out string name);
+		internal static extern ErrorCode gp_list_get_name (HandleRef list, int index, out IntPtr name);
 
 		public string GetName (int index)
 		{
-			string name;
+			IntPtr name;
 
 			Error.CheckError (gp_list_get_name(this.Handle, index, out name));
 
-			return name;
+			return Marshal.PtrToStringAnsi (name);
 		}
 		
 		[DllImport ("libgphoto2.so")]
-		internal static extern ErrorCode gp_list_get_value (HandleRef list, int index, out string value);
+		internal static extern ErrorCode gp_list_get_value (HandleRef list, int index, out IntPtr value);
 
 		public string GetValue (int index)
 		{
-			string value;
+			IntPtr value;
 
 			Error.CheckError (gp_list_get_value(this.Handle, index, out value));
 
-			return value;
+			return Marshal.PtrToStringAnsi (value);
 		}
 		
 		[DllImport ("libgphoto2.so")]
